### PR TITLE
Close navbar after clicking outside of it, close #24

### LINKED ISSUE #24: Close navbar after clicking outside of it

### PR DESCRIPTION
**Pull request by ganneo**

### CODE REVIEW

File: src/streak.js [hunk 1]
```diff
@@ -18,3 +19,17 @@ wordService.get().then((words) => {
...
+    document.getElementById("streak-main").addEventListener("click", () => {
+      document.getElementById("hamburger").click();
+    });
+  } else {
+    navOpen = false;
+    document.getElementById("streak-main").removeEventListener("click", () => {
+      click(document.getElementById("hamburger"));
+    });
+  }
+});
```
Comment (tiffanygan) [inline src/streak.js]: This is a very good function, but repeating this in every javascript file seems a bit redundant. Is there a way you can share the same function over files?

File: src/register.js [hunk 2]
```diff
@@ -1,10 +1,12 @@
+import Util from "./lib/Util";
 import { CURR_USER_KEY } from "./Constant";
 import RegisterUI from "./ui/RegisterUI";
 import UserService from "./service/UserService";
 
 const ui = new RegisterUI(document);
 const userService = new UserService();
 let registerSuccess = null;
+let navOpen = false;
```
Comment (tiffanygan) [inline src/register.js]: You don't need this anymore because you imported the function from `Util`

File: src/register.js [hunk 3]
```diff
@@ -41,3 +43,19 @@ ui.registerBtn.addEventListener("click", async (e) => {
...
+    document.getElementById("register-main").addEventListener("click", () => {
+      document.getElementById("hamburger").click();
+    });
+  } else {
+    navOpen = false;
+    document.getElementById("register-main").removeEventListener("click", () => {
+      click(document.getElementById("hamburger"));
+    });
+  }
+});
```
Comment (tiffanygan) [inline src/register.js]: same as line 9

**Review by tiffanygan — CHANGES_REQUESTED** [review]

**Review by tiffanygan — APPROVED** [review]

File: src/lib/Util.js [hunk 4]
```diff
@@ -32,4 +32,22 @@ export default class Util {
...
+          hamburgerBtn.click();
+        });
+      } else {
+        navOpen = false;
+        mainElement.removeEventListener("click", () => {
+          click(hamburgerBtn);
+        });
+      }
+    });
+  }
```
Comment (yugan2005) [inline src/lib/Util.js]: This logic need be discussed. 
1. We should not `addEventListener` and `removeEventListener` repetitively.
2. Instead, we should use event delegation
3. We can check the attribute to get whether the navbar is collapsed or not

File: src/model/User.js [hunk 5]
```diff
@@ -37,6 +34,11 @@ export default class User {
     if (this.streakArray[this.streakArray.length - 1] !== date.toISOString()) {
       this.streakArray.push(date.toISOString());
 
+      this.streak = 1;
+      this.longestStreak = 1;
+    }
+
+    if (this.streakArray[this.streakArray.length - 1] !== date) {
```
Comment (tiffanygan) [inline src/model/User.js]: You don't need these lines

**Review by tiffanygan — CHANGES_REQUESTED** [review]

**Review by tiffanygan — APPROVED** [review]

**Pull request merged**